skimage/_shared/fast_exp.h

```diff
@@ -62,12 +90,12 @@ __inline float fast_expf (float y)
     switch(_eco.t[0]) {
         case 1:
             /* Little endian */
-            _eco.n.j = (int)(EXPF_A*(y)) + EXPF_BC;
+            _eco.n.j = (int16_t)(EXPF_A*(y)) + EXPF_BC;
```

Comment:
:+1: 
I'd use `int32_t` explicitly in the original function as well (in a separate commit, obviously).